src/cljstyle/format/core.clj

```diff
@@ -166,6 +166,25 @@
     [nil text]))
 
 
+(defn- trim-trailing-newlines
+  "Trims the appropriate number of newlines (either of all of them or all
+  except one) from the result of formatting a file."
+  [formatted-text rules-config]
+  (let [num-trailing-blanks
+        (-> formatted-text
+            (reverse)
+            (->> (map-indexed vector)
+                 (filter (fn [x] (not= \newline (second x)))))
+            (first)
+            (first)
+            (cond->
+              (get-in rules-config [:eof-newline :enabled?])
+              (dec)))]
+    (subs formatted-text
+          0
+          (- (count formatted-text) num-trailing-blanks))))
```

Comment:
A simpler approach to this might be to use a regex replacement:
```clojure
=> (str/replace-first "foo\n\n\n\n\n" #"\n*\z" "\n")
"foo\n"

=> (str/replace-first "foo\n" #"\n*\z" "\n")
"foo\n"

=> (str/replace-first "foo" #"\n*\z" "\n")
"foo\n"
```

Reply:
yeah that's much cleaner